src/mixings/base_mixing.hpp

```diff
@@ -37,6 +42,12 @@ class BaseMixing {
   virtual double mass_new_cluster(const unsigned int n_clust,
                                   const unsigned int n) const = 0;
 
+  virtual void update_hypers(
+      const std::vector<std::shared_ptr<BaseHierarchy>> &unique_values,
+      unsigned int n) = 0;
+
+  virtual void set_state(google::protobuf::Message *out) = 0;
```

Comment:
I would pass a const reference here, since it's setting the state of the Mixing and not writing anything in 'out'

Also observe that polymorphism / casting works identical with references

Reply:
I think I tried a const ref and it didn't work. I'll try again

Reply:
Depends on what you're casting to: you should downcast to a const reference to DPState